### PR TITLE
#53115 Add getCategory method

### PR DESCRIPTION
**Pull request by Ramiz4ik**

### CODE REVIEW

File: Block/NodeType/Category.php [hunk 1]
```diff
@@ -150,6 +159,24 @@ public function getCategoryUrl($nodeId, $storeId = null)
...
+    {
+        if (!isset($this->nodes[$nodeId])) {
+            throw new \InvalidArgumentException('Invalid node identifier specified');
+        }
+
+        $storeId = $this->_storeManager->getStore()->getId();
+        $node = $this->nodes[$nodeId];
+        $categoryId = (int) $node->getContent();
+
+        return $this->categoryRepository->get($categoryId, $storeId);
```
Comment (dariuszmaciejewski) [inline Block/NodeType/Category.php]: Do you load a category in a loop here (which renders a single node)? I think isn't the best idea. Can we load all needed categories as a collection before rendering our menu?

Comment (Ramiz4ik) [reply Block/NodeType/Category.php]: Done

File: Model/NodeType/Category.php [hunk 2]
```diff
@@ -97,16 +106,26 @@ public function fetchData(array $nodes, $storeId)
...
         $this->profiler->stop(__METHOD__);
 
-        return [$localNodes, $categoryUrls];
+        return [$localNodes, $categoryUrls, $categories];
+    }
+
+    public function getCategory($categoryId)
+    {
+        $category = $this->categoryFactory->create();
+        $category->load($categoryId);
```
Comment (dariuszmaciejewski) [inline Model/NodeType/Category.php]: As I said before - use a category collection to load all needed categories from DB at once.

File: Model/NodeType/Category.php [hunk 3]
```diff
@@ -97,16 +106,26 @@ public function fetchData(array $nodes, $storeId)
 
         $localNodes = [];
         $categoryIds = [];
+        $categories = [];
 
         foreach ($nodes as $node) {
             $localNodes[$node->getId()] = $node;
-            $categoryIds[] = (int)$node->getContent();
+            $categoryId = (int)$node->getContent();
+            $categoryIds[] = $categoryId;
```
Comment (dariuszmaciejewski) [inline Model/NodeType/Category.php]: You can use `array_keys($categories)` instead later.

File: Model/NodeType/Category.php [hunk 4]
```diff
@@ -104,9 +113,33 @@ public function fetchData(array $nodes, $storeId)
...
+        $categories = $this->getCategories($storeId, $categoryIds);
 
         $this->profiler->stop(__METHOD__);
 
-        return [$localNodes, $categoryUrls];
+        return [$localNodes, $categoryUrls, $categories];
+    }
+
+    /**
+     * @param int $storeId
```
Comment (jochhop) [inline Model/NodeType/Category.php]: From what I see in `setStore` method definition, the method accepts all these types:

```php
@param int|string|\Magento\Store\Model\Store $store
```

So please update parameter name to be `$store` instead and add the accepted types to the annotations (not only `int`)

Comment (Ramiz4ik) [reply Model/NodeType/Category.php]: Done

**Pull request merged**